segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/stat/SegmentStatsRecorderTest.java

```diff
@@ -181,6 +181,31 @@ public void testMetrics() {
         assertEquals(NO_COUNTER_VALUE, getCounterValue(MetricsNames.SEGMENT_READ_BYTES, segmentName));
     }
 
+    @Test(timeout = 10000)
+    public void testTransactionMetrics() throws InterruptedException {
+        val segmentName = getStreamSegmentName();
+
+        long dataLength = 321;
+        int interval = 500;
+        val duration = Duration.ofMillis(2 * interval);
+        @Cleanup
+        val context = new TestContext(segmentName, duration, false);
+
+        context.statsRecorder.createSegment(segmentName, ScalingPolicy.ScaleType.BY_RATE_IN_KBYTES_PER_SEC.getValue(), 2, Duration.ofSeconds(1));
+        context.statsRecorder.recordAppend(segmentName, dataLength, 5, Duration.ofSeconds(2));
+        val txnName1 = NameUtils.getTransactionNameFromId(segmentName, UUID.randomUUID());
+        context.statsRecorder.createSegment(txnName1, ScalingPolicy.ScaleType.BY_RATE_IN_KBYTES_PER_SEC.getValue(), 2, Duration.ofSeconds(1));
+        Thread.sleep(interval);
```

Comment:
Please do not put Thread.sleep in your tests. Rework your test or use mocks to simulate this.